packages/shopify-codemod/transforms/split-return-assignments.js

```diff
@@ -0,0 +1,52 @@
+import {matchLast, forEachAssignment} from './utils';
+
+export default function splitReturnAssignments({source}, {jscodeshift: j}, {printOptions = {}}) {
+  const sourceAST = j(source);
+  sourceAST
+    .find(j.Function, {
+      body: {
+        body: matchLast(j.ReturnStatement, {
+          type: 'ReturnStatement',
+          argument: j.AssignmentExpression,
+        }),
+      },
+    })
+    .forEach(({node: {body: {body}}}) => {
+      const returnLine = body[body.length - 1];
+      forEachAssignment(body[body.length - 1].argument, (assignment) => {
```

Comment:
You still haven't addressed the comment I made about disliking the "callback-for-each-item" style being used here.